libs/ui-components/src/components/Table/TablePagination.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Trans } from 'react-i18next';
+import { Pagination, PaginationToggleTemplateProps, PaginationVariant, Spinner } from '@patternfly/react-core';
+
+import { PAGE_SIZE } from '../../constants';
+import { useTranslation } from '../../hooks/useTranslation';
+
+type TablePaginationProps = {
+  isUpdating: boolean;
+  itemCount?: number;
+  currentPage: number;
+  setCurrentPage: (page: number) => void;
+};
+
+const PaginationTemplate = ({
+  isUpdating,
+  firstIndex,
+  itemCount,
+}: PaginationToggleTemplateProps & { isUpdating: boolean }) => {
+  const { t } = useTranslation();
+
+  if (isUpdating) {
+    return <Spinner size="sm" />;
+  }
+
+  const totalPages = String(Math.ceil((itemCount || 0) / PAGE_SIZE));
+  const currentPage = String(Math.ceil((firstIndex || 0) / PAGE_SIZE));
+  return (
+    <Trans t={t}>
+      {currentPage} of <strong>{totalPages}</strong>
+    </Trans>
+  );
+};
+
+const TablePagination = ({ isUpdating, itemCount, currentPage, setCurrentPage }: TablePaginationProps) => {
+  const { t } = useTranslation();
+
+  const onSetPage = (_event: React.MouseEvent | React.KeyboardEvent | MouseEvent, newPage: number) => {
+    setCurrentPage(newPage);
+  };
+
+  return (
+    <Pagination
```

Comment:
clicking fast on back/forward messes up the pagination. It won't change the content of the page. Maybe we should disable these buttons while fetching ?

Reply:
I also added an improvement to avoid showing only the spinner while the new page is loading.
Now we'll show the loader and also the page number control.